packages/authorization-process/src/model/domain/errors.ts

```diff
@@ -103,3 +112,73 @@ export function userAlreadyAssigned(
     title: "user Already Assigned",
   });
 }
+
+export function eserviceNotFound(eserviceId: EServiceId): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `EService ${eserviceId} not found`,
+    code: "eserviceNotFound",
+    title: "EService not found",
+  });
+}
+
+export function purposeNotFound(purposeId: PurposeId): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `Purpose ${purposeId} not found`,
+    code: "purposeNotFound",
+    title: "Purpose not found",
+  });
+}
+
+export function noVersionsFoundInPurpose(
+  purposeId: PurposeId
+): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `No versions found in purpose ${purposeId}`,
+    code: "noVersionsFoundInPurpose",
+    title: "No versions found in purpose",
+  });
+}
+
+export function descriptorNotFound(
+  eserviceId: EServiceId,
+  descriptorId: DescriptorId
+): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `Descriptor ${descriptorId} not found in EService ${eserviceId}`,
+    code: "descriptorNotFound",
+    title: "Descriptor not found",
+  });
+}
+
+export function agreementNotFound(
+  eserviceId: EServiceId,
+  consumerId: TenantId
+): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `Agreement not found for eservice ${eserviceId} and consumer ${consumerId}`,
+    code: "agreementNotFound",
+    title: "Agreement not found",
+  });
+}
```

Comment:
```suggestion
export function noAgreementFoundInRequiredState(
  eserviceId: EServiceId,
  consumerId: TenantId
): ApiError<ErrorCodes> {
  return new ApiError({
    detail: `No agreement in required state found for eservice ${eserviceId} and consumer ${consumerId}`,
    code: "noAgreementFoundInRequiredState",
    title: "No Agreement found in required state",
  });
}
```
or something like that

Reply:
Updated